lib/xml_security.rb

```diff
@@ -255,12 +255,12 @@ def validate_document_with_cert(idp_cert)
         begin
           cert = OpenSSL::X509::Certificate.new(cert_text)
         rescue OpenSSL::X509::CertificateError => _e
-          return append_error("Certificate Error", soft)
+          return append_error("Document certificate error", soft)
         end
 
         # check saml response cert matches provided idp cert
         if idp_cert.to_pem != cert.to_pem
-          return false
+          return append_error("Document certificate does not match idp certificate", soft)
```

Comment:
In an environment where multiple idps cert are registered , that error gonna be registered for each validation executed.
In my opinion, this error should not be registered here because can't confuse the admin reviewing the logs.

I'm ok raising an error that none of the registered certs matched the certificate in the document which will require some refactor in the code,

Reply:
Please let me know what you think about the new chagnes